### PR TITLE
Optimize model library

### PR DESCRIPTION
**Pull request by hayden-fr**
Adjust model list responses to prepare optimized model library.

Add model preview interface.

### CODE REVIEW

File: server.py [hunk 1]
```diff
@@ -221,20 +222,6 @@ async def get_root(request):
         def get_embeddings(self):
             embeddings = folder_paths.get_filename_list("embeddings")
             return web.json_response(list(map(lambda a: os.path.splitext(a)[0], embeddings)))
-        
-        @routes.get("/models")
```
Comment (huchenlei) [inline server.py:224]: We need to keep these routes for now if we want to merge this PR first. If we decide to land this PR after frontend changes has landed, the frontend side needs to be able to handle both APIs.

**Review by huchenlei — CHANGES_REQUESTED** [review]

File: server.py (hunk 1 above)
Comment (hayden-cn) [reply server.py:224]: These two APIs have not been removed, just moved to `model_filemanager/model_filemanager.py`. I can restore it if needed.

Comment (hayden-cn) [reply server.py:224]: The new API adds the prefix of `/experiment`. My plan is to retain the original API and then migrate it to the original API after the new function is stable.

**Review by huchenlei — CHANGES_REQUESTED** [review]
Can you move model_filemanager.py to `app/`. All server extra paths should go there.

**Comment by hayden-fr** [comment]
> Can you move model_filemanager.py to `app/`. All server extra paths should go there.

Ok

**Review by huchenlei — APPROVED** [review]

**Pull request merged**